framework/src/modules/chain/bft/finality_manager.js

```diff
@@ -290,6 +345,26 @@ class FinalityManager {
 		return true;
 	}
 
+	/**
+	 * Cleanup pre-votes and pre-commits objects
+	 * to only keep the track of maximum 5 rounds (headers list size)
+	 *
+	 * @private
+	 */
+	_cleanup() {
```

Comment:
Is this slicing only one round or 505 headers?
Worth mentioning in the description!

Reply:
505 headers, I will add few comments further.

Reply:
Its mentioned in the function description in comments that its cleaning 5 rounds.